### PR TITLE
UCP/FLUSH: Fix error handling from UCT EP flush

### LINKED ISSUE #7742: failure in tcp/test_ucp_sockaddr.concurrent_disconnect/1  on r-vmb-ppc-jenkins
### Describe the bug
happened here http://hpc-master.lab.mtl.com:8080/blue/rest/organizations/jenkins/pipelines/ucx/runs/15412/nodes/38/steps/259/log/?start=0
```
[2021-12-02T12:35:34.118Z] [ RUN      ] tcp/test_ucp_sockaddr.concurrent_disconnect/1 <tcp/tag,mt>
[2021-12-02T12:35:34.376Z] [     INFO ] Testing 10.209.45.93:0
[2021-12-02T12:35:34.376Z] [     INFO ] server listening on 10.209.45.93:45483
[2021-12-02T12:36:42.198Z] /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.cc:866: Failure
[2021-12-02T12:36:42.198Z] Expected: (UCS_INPROGRESS) != (status), actual: 1 vs 1
[2021-12-02T12:36:42.198Z] free not completed EP close request: 0x1005472bd18
[2021-12-02T12:51:34.334Z] /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/test_helpers.cc:55: Failure
[2021-12-02T12:51:34.334Z] Failed
[2021-12-02T12:51:34.334Z] Connection timed out - abort testing
[2021-12-02T12:51:34.334Z] [r-vmb-ppc-jenkins:31022:0:31022] Caught signal 6 (Aborted: tkill(2) or tgkill(2))
[2021-12-02T12:51:35.264Z] 
[2021-12-02T12:51:35.264Z] /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/sys/event_set.c: [ ucs_event_set_wait() ]
[2021-12-02T12:51:35.264Z]       ...
[2021-12-02T12:51:35.264Z]       195 
[2021-12-02T12:51:35.264Z]       196     events = ucs_alloca(sizeof(*events) * *num_events);
[2021-12-02T12:51:35.264Z]       197 
[2021-12-02T12:51:35.264Z] ==>   198     nready = epoll_wait(event_set->event_fd, events, *num_events, timeout_ms);
[2021-12-02T12:51:35.264Z]       199     if (ucs_unlikely(nready < 0)) {
[2021-12-02T12:51:35.264Z]       200         *num_events = 0;
[2021-12-02T12:51:35.264Z]       201         if (errno == EINTR) {
[2021-12-02T12:51:35.264Z] 
[2021-12-02T12:51:35.520Z] ==== backtrace (tid:  31022) ====
[2021-12-02T12:51:35.520Z]  0 0x000000000007a3dc ucs_event_set_wait()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/sys/event_set.c:198
[2021-12-02T12:51:35.520Z]  1 0x000000000003284c uct_tcp_iface_progress()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/tcp/tcp_iface.c:327
[2021-12-02T12:51:35.520Z]  2 0x000000000005f124 ucs_callbackq_dispatch()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucs/datastruct/callbackq.h:211
[2021-12-02T12:51:35.520Z]  3 0x000000000005f124 uct_worker_progress()  /scrap/jenkins/workspace/ucx-3/contrib/../src/uct/api/uct.h:2589
[2021-12-02T12:51:35.520Z]  4 0x000000000005f124 ucp_worker_progress()  /scrap/jenkins/workspace/ucx-3/contrib/../src/ucp/core/ucp_worker.c:2664
[2021-12-02T12:51:35.520Z]  5 0x00000000107ff6e8 ucp_test_base::entity::progress()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.cc:1017
[2021-12-02T12:51:35.520Z]  6 0x0000000010804450 ucp_test::progress()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.cc:168
[2021-12-02T12:51:35.520Z]  7 0x0000000010804450 ucp_test::progress()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.cc:176
[2021-12-02T12:51:35.520Z]  8 0x0000000010804450 ucp_test_base::entity::close_all_eps()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.cc:888
[2021-12-02T12:51:35.520Z]  9 0x000000001080466c ucp_test::disconnect()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.cc:226
[2021-12-02T12:51:35.520Z] 10 0x00000000108046f0 ucp_test::cleanup()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.cc:86
[2021-12-02T12:51:35.520Z] 11 0x0000000010234130 ucs::test_base::TearDownProxy()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/test.cc:324
[2021-12-02T12:51:35.520Z] 12 0x00000000104a20d0 ucp_test::TearDown()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/ucp/ucp_test.h:192
[2021-12-02T12:51:35.520Z] 13 0x0000000010b567f0 HandleSehExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:2433
[2021-12-02T12:51:35.520Z] 14 0x0000000010b567f0 testing::internal::HandleExceptionsInMethodIfSupported<testing::Test, void>()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:2469
[2021-12-02T12:51:35.520Z] 15 0x0000000010b49a70 testing::Test::Run()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:2517
[2021-12-02T12:51:35.520Z] 16 0x0000000010b49cf8 testing::TestInfo::Run()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:2687
[2021-12-02T12:51:35.520Z] 17 0x0000000010b49e54 testing::TestSuite::Run()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:2819
[2021-12-02T12:51:35.520Z] 18 0x0000000010b4aed0 testing::internal::UnitTestImpl::RunAllTests()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:5350
[2021-12-02T12:51:35.520Z] 19 0x0000000010b4b164 HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:2433
[2021-12-02T12:51:35.520Z] 20 0x0000000010b4b164 HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl, bool>()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:2469
[2021-12-02T12:51:35.520Z] 21 0x0000000010b4b164 testing::UnitTest::Run()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.cc:4940
[2021-12-02T12:51:35.520Z] 22 0x00000000101a3890 RUN_ALL_TESTS()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/googletest/gtest.h:2473
[2021-12-02T12:51:35.520Z] 23 0x00000000101a3890 main()  /scrap/jenkins/workspace/ucx-3/contrib/../test/gtest/common/main.cc:118
[2021-12-02T12:51:35.520Z] 24 0x0000000000025100 generic_start_main.isra.0()  libc-start.c:0
[2021-12-02T12:51:35.520Z] 25 0x00000000000252f4 __libc_start_main()  ???:0
[2021-12-02T12:51:35.520Z] =================================
```

### PR DESCRIPTION
**Pull request by dmitrygx**
## What

Fix error handling from UCT EP flush.

## Why ?

Fixes #7742
UCT/TCP endpoints could return an error code from `uct_ep_flush()` and it should decrease completion counter set in UCP flush request prior calling `ucp_ep_flush_progress()` and checking flush completion, otherwise - it may stuck.

## How ?

1. Fix code style in `ucp_ep_flush_progress_pending()` function.
2. Move marking of a UCP flush request as failed where handling of `UCS_OK` to decrease completion counter in a UCP request prior checking completion.
3. Reorganize checking error code at the end of the function to simplify it.
4. Add logging of UCT EP pointer - it could be useful to find out the exact UCT EP which is completed with some status.

### CODE REVIEW

**Comment by dmitrygx** [comment]
@brminich @hoopoepg could you review pls?

File: src/ucp/rma/flush.c [hunk 1]
```diff
@@ -225,18 +227,16 @@ ucs_status_t ucp_ep_flush_progress_pending(uct_pending_req_t *self)
...
             req->send.lane = UCP_NULL_LANE;
         }
-        return UCS_OK;
-    } else if (status == UCS_ERR_NO_RESOURCE) {
-        return UCS_ERR_NO_RESOURCE;
-    } else {
-        ucp_ep_flush_error(req, status);
-        return UCS_OK;
     }
+
```
Comment (hoopoepg) [inline src/ucp/rma/flush.c:238]: removed ucp_ep_flush_error?

Comment (dmitrygx) [reply src/ucp/rma/flush.c:238]: yes, it was moved above, it has to be done prior `ucp_ep_flush_progress` and checking completion of the request

**Review by hoopoepg — APPROVED** [review]

**Comment by dmitrygx** [comment]
@yosefe could you review pls?
wdyt about porting it to v1.12?

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]
good catch and fix

**Comment by dmitrygx** [comment]
@yosefe thanks! do we need this for v1.12.x?

**Comment by yosefe** [comment]
> @yosefe thanks! do we need this for v1.12.x?

I think yes. @brminich WDYT?

**Comment by brminich** [comment]
imo, it is simple enough to be ported

**Pull request merged**